infra/couchdb/verify-access-model.sh

```
@@ -0,0 +1,105 @@
+  curl -s -u "$ADMIN" -X PUT "$URL/_users/org.couchdb.user:$u" \
+    -H 'Content-Type: application/json' \
+    -d "{\"name\":\"$u\",\"password\":\"$PW\",\"roles\":[],\"type\":\"user\"}" >/dev/null
+done
+
+curl -s -u "$ADMIN" -X PUT "$URL/$DB/_security" -H 'Content-Type: application/json' \
+  -d '{"members":{"names":["vam-writer","vam-reader"],"roles":[]},"writers":{"names":["vam-writer"]}}' >/dev/null
+
+# The design doc mirrors infra/couchdb/design-docs/access.js. Keep them in step.
+curl -s -u "$ADMIN" -X PUT "$URL/$DB/_design/access" -H 'Content-Type: application/json' -d '{
+  "validate_doc_update": "function (newDoc, oldDoc, userCtx, secObj) { if (userCtx.roles.indexOf(\"_admin\") !== -1) { return } var writers = (secObj && secObj.writers && secObj.writers.names) || []; if (writers.indexOf(userCtx.name) === -1) { throw { forbidden: \"You have read-only access to this project.\" } } if (newDoc._deleted) { return } if (!newDoc.type) { throw { forbidden: \"Every document must carry a `type` field.\" } } if (newDoc.type === \"audit\" && oldDoc) { throw { forbidden: \"Audit entries are immutable.\" } } }"
+}' >/dev/null
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)